src/JsonLd/ContextBuilder.php

```diff
@@ -97,18 +97,25 @@ public function getResourceContext(string $resourceClass, int $referenceType = U
             }
 
             $convertedName = $this->nameConverter ? $this->nameConverter->normalize($propertyName) : $propertyName;
+            $jsonldContext = $propertyMetadata->getAttributes()['jsonld_context'] ?? [];
 
             if (!$id = $propertyMetadata->getIri()) {
                 $id = sprintf('%s/%s', $prefixedShortName, $convertedName);
             }
 
             if (true !== $propertyMetadata->isReadableLink()) {
-                $context[$convertedName] = [
+                $jsonldContext = [
                     '@id' => $id,
                     '@type' => '@id',
-                ];
-            } else {
+                ] + $jsonldContext;
```

Comment:
Ok, I thought in this case it might be better to force @id and @type. But the other way round it leaves more freedom for the developers. I'll change it.